### PR TITLE
Added option to allow disabling TLS v1.3

### PR DESCRIPTION
**Pull request by mayank3012jain**
Changed the maxVersion when TLSv1.3 is to be disable, since secureOptions does not support it in node 12

### CODE REVIEW

File: CHANGELOG.yaml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+master:
+  new features:
+    - GH-1130 Added support to disable TLSv1.3
```
Comment (codenirvana) [inline CHANGELOG.yaml:3]: where?
> ... in `tlsDisabledProtocols` protocol profile behavior.

File: lib/requester/core.js [hunk 2]
```diff
@@ -439,7 +439,14 @@ module.exports = {
         // the SSL and TLS protocol versions to disabled during negotiation
         if (Array.isArray(protocolProfileBehavior.tlsDisabledProtocols)) {
             protocolProfileBehavior.tlsDisabledProtocols.forEach(function (protocol) {
-                options.secureOptions |= constants[SSL_OP_NO + protocol];
+                // since secure options doesn't support TLSv1.3 before Node 14
+                // @todo remove the if condition when we drop support for Node 12
+                if (protocol === 'TLSv1_3' && !constants[SSL_OP_NO + protocol]) {
+                    options.maxVersion = 'TLSv1.2';
```
Comment (codenirvana) [inline lib/requester/core.js:445]: Since tests didn't cover this, have you ensure this works in Node v14?

File: lib/requester/core.js [hunk 3]
```diff
@@ -439,7 +439,14 @@ module.exports = {
...
+                }
+                else {
+                    options.secureOptions |= constants[SSL_OP_NO + protocol];
```
Comment (codenirvana) [inline lib/requester/core.js:448]: Why is this not covered by tests 🤔

File: test/integration/protocol-profile-behavior/tlsProtocols.test.js [hunk 4]
```diff
@@ -977,7 +978,7 @@ var fs = require('fs'),
                 });
             });
 
-            (TLSv1_3_SUPPORTED ? describe.skip : describe)('with just SSLv23 enabled', function () {
+            (TLSv1_3_SUPPORTED ? describe : describe)('with just SSLv23 enabled', function () {
```
Comment (codenirvana) [inline test/integration/protocol-profile-behavior/tlsProtocols.test.js:981]: Don't think this condition is needed anymore.

**Review by codenirvana — CHANGES_REQUESTED** [review]

File: lib/requester/core.js (hunk 2 above)
Comment (mayank3012jain) [reply lib/requester/core.js:445]: Yes, checked in Node v14. Works correctly without the `if` condition.

File: lib/requester/core.js (hunk 3 above)
Comment (mayank3012jain) [reply lib/requester/core.js:448]: There are tests that cover this. But the codecov build is failing. Still trying to figure out why this is happening.

File: CHANGELOG.yaml [hunk 5]
```diff
@@ -1,3 +1,7 @@
+master:
+  new features:
+    - GH-1130 Added support to disable TLSv1.3 in tlsDisabledProtocols protocol profile behavior.
```
Comment (codenirvana) [inline CHANGELOG.yaml:3]:
```suggestion
    - GH-1130 Added support to disable TLSv1.3 in tlsDisabledProtocols protocol profile behavior
```

**Review by codenirvana — CHANGES_REQUESTED** [review]

**Review by codenirvana — DISMISSED** [review]

**Review by codenirvana — APPROVED** [review]

**Pull request merged**